tests/e2e/lib.sh

```diff
@@ -58,56 +56,65 @@ deploy_stackrox_with_custom_central_and_sensor_versions() {
     # and since latest sensor is not available in the repo either the chart is created here in the elif case.
+    chart_parent_dir="./sensor-chart-dir-compatibility-tests"
+    chart_name="stackrox-secured-cluster-services"
     if [[ $helm_charts =~ $sensor_regex ]]; then
-        sensor_chart="${helm_repo_name}/stackrox-secured-cluster-services"
-        ci_export SENSOR_CHART_DIR_OVERRIDE "./sensor-chart-dir-compatibility-tests"
-        helm pull "${sensor_chart}" --version "${SENSOR_CHART_VERSION_OVERRIDE}" --untar --untardir "${SENSOR_CHART_DIR_OVERRIDE}"
-    elif [[ "$latest_tag" == "$SENSOR_CHART_VERSION_OVERRIDE" ]]; then
-        if [[ $(roxctl version) != "$latest_tag" ]]; then
-            echo >&2 "Roxctl version $(roxctl version) is not equal to latest tag ${latest_tag}"
+        sensor_chart="${helm_repo_name}/${chart_name}"
+        ci_export SENSOR_CHART_DIR_OVERRIDE "${chart_parent_dir}/${chart_name}"
+        helm pull "${sensor_chart}" --version "${SENSOR_CHART_VERSION_OVERRIDE}" --untar --untardir "${chart_parent_dir}"
+        echo "Pulled helm chart for ${chart_name} to ${SENSOR_CHART_DIR_OVERRIDE}, running ls -a"
+        set -x
+        ls -la "${SENSOR_CHART_DIR_OVERRIDE}"
+        set +x
+    elif [[ "$current_tag" == "$SENSOR_CHART_VERSION_OVERRIDE" ]]; then
+        if [[ $(roxctl version) != "$current_tag" ]]; then
+            >&2 echo "Reported roxctl version "$(roxctl version)" is different from requested tag ${current_tag}. It won't be possible to get helm charts for ${current_tag}. Please check test setup."
             exit 1
         fi
-        ci_export SENSOR_CHART_DIR_OVERRIDE "./sensor-chart-dir-compatibility-tests"
+        ci_export SENSOR_CHART_DIR_OVERRIDE "${chart_parent_dir}/${chart_name}"
         roxctl helm output secured-cluster-services --image-defaults=opensource --output-dir "${SENSOR_CHART_DIR_OVERRIDE}" --remove
-        echo "Downloaded stackrox-secured-cluster-services helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} to ${SENSOR_CHART_DIR_OVERRIDE}"
-        unset SENSOR_CHART_VERSION_OVERRIDE
+        echo "Downloaded ${chart_name} helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} to ${SENSOR_CHART_DIR_OVERRIDE}"
     else
-        echo >&2 "stackrox-secured-cluster-services helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} not found in ${helm_repo_name} repo nor is it the latest tag."
+        echo >&2 "${chart_name} helm chart for version ${SENSOR_CHART_VERSION_OVERRIDE} not found in ${helm_repo_name} repo nor is it the latest tag."
         exit 1
     fi
 
     deploy_stackrox
 
     if [[ -n "${CENTRAL_CHART_DIR_OVERRIDE}" ]]; then
-        rm -r "${CENTRAL_CHART_DIR_OVERRIDE}"
+        rm -rf "${CENTRAL_CHART_DIR_OVERRIDE}"
     fi
     if [[ -n "${SENSOR_CHART_DIR_OVERRIDE}" ]]; then
-        rm -r "${SENSOR_CHART_DIR_OVERRIDE}"
+        rm -rf "${SENSOR_CHART_DIR_OVERRIDE}"
     fi
```

Comment:
Looking at this now, I think it can be simplified.

First, before pulling any charts, you can create a **parent** directory
```bash
charts_dir="$(mktemp -d ./charts-dir.XXXXXX)"
```

Next, do pull Central and Sensor charts, install stackrox.

Finally, instead of `if [[ -n ... ]]; then rm -rf ...` you can simply
```bash
rm -rf "$charts_dir"
```

The thing is: the parent directory can even stay empty in case no overrides are applied (not our case but still), but it is **created** by `mktemp` and so you can delete it **unconditionally** in the end. Everything inside will be deleted too.

WDYT?

Reply:
`mktemp -d` ensures the name is unique so you don't have to worry about it.